storey/flow.py

```diff
@@ -170,11 +174,38 @@ async def _run_loop(self):
         try:
             for path in self._paths:
                 async with aiofiles.open(path, mode='r') as f:
-                    if self._skip_header:
-                        await f.readline()
+                    header = None
+                    if self._with_header:
+                        line = await f.readline()
+                        if self._build_dict:
+                            header = next(csv.reader([line]))
                     async for line in f:
                         parsed_line = next(csv.reader([line]))
-                        await self._do_downstream(Event(parsed_line, None, datetime.now(), None))
+                        element = parsed_line
+                        key = None
+                        timestamp = None
+                        if header:
+                            if len(parsed_line) != len(header):
+                                raise ValueError(f'CSV line with {len(parsed_line)} fields did not match header with {len(header)} fields')
+                            element = {}
+                            for i in range(len(parsed_line)):
+                                if header:
+                                    element[header[i]] = parsed_line[i]
+                                else:
+                                    element[i] = parsed_line[i]
+                        if self._key_field:
+                            key = element[self._key_field]
```

Comment:
looks like you assume that `key_field` and `timestamp_field` will always be used with `build_dict`. not sure thats a correct assumtption.

Reply:
I don't. They will need to indicate the position in that case.

Reply:
well.. that's not intuitive at all
i'd assume the user can either specify a position or a column name and you'll have manage either way.
you can get the position based on the column name..

Reply:
What about when there is no header?